proposals/kubeflow-steering-committee-election-proposal.md

```diff
@@ -0,0 +1,125 @@
+The table below illustrates the cohorts and election schedules:
+
+<table>
+  <thead>
+    <tr>
+      <th>Timeline</th>
+      <th>2-member cohort</th>
+      <th>3-member cohort</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>2023</td>
+      <td>Interim KSC member A<br>
+Interim KSC member B</td>
+      <td>New election</td>
+    </tr>
+    <tr>
+      <td>2024</td>
+      <td>New election</td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>2025</td>
+      <td></td>
+      <td>New election</td>
+    </tr>
+    <tr>
+      <td>2026</td>
+      <td>New election</td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>2027</td>
+      <td></td>
+      <td>New election</td>
+    </tr>
+    <tr>
+      <td>…</td>
+      <td></td>
+      <td></td>
+    </tr>
+  </tbody>
+</table>
+
+## Election Procedure
+
+### Timeline ###  
+Steering Committee elections are held annually. 4 weeks or more before the election, the Steering Committee will appoint Election Officer(s) (see below). 2 weeks or more before the election, the Election Officer(s) will issue a call for nominations, publish the list of eligible voters, and open the call for exceptions. One week before the election the call for nominations and exceptions will be closed. The election will be open for voting not less than two weeks and not more than four. The results of the election will be announced within one week of closing the election. New Steering Committee members will take office in December of each year on the date the results are announced.
+
+### Election Officer(s) ###  
+4 weeks or more before the election, the Steering Committee will appoint between one and three Election Officer(s) to administer the election. Elections Officers will be Kubeflow community members in good standing who are eligible to vote, are not running for Steering in that election, who are not currently part of the Steering Committee and can make a public promise of impartiality. They will be responsible for:
+
+-  Making all announcements associated with the election
+-  Preparing and distributing electronic ballots
+-  Judging exception requests
+-  Assisting candidates in preparing and sharing statements
+-  Tallying voting results according to the rules in this charter
+
+### Eligibility to Vote ###
+Anyone who has at least 50 contributions in the last 12 months is eligible to vote in the Steering election. Contributions are defined as opening PRs, reviewing and commenting on PRs, opening and commenting on issues, writing design docs, commenting on design docs, helping people on slack, participating in working groups, and other efforts that help advance the Kubeflow project.
```

Comment:
Also, as raised by @juliusvonkohout in https://github.com/kubeflow/community/pull/645#issuecomment-1697668212, I think putting a minimum of 1 merged PR (even a minor docs update is sufficient) will help prevent spam.

Also, as I raised in https://github.com/kubeflow/community/pull/645#issuecomment-1697886798, this will additionally ensure the member has signed the CLA.

Reply:
Borrowed this query from kNative election eligiblity: https://kubeflow.devstats.cncf.io/d/9/developer-activity-counts-by-repository-group-table?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All

Currently there are 46 eligible people meeting the 50 cutoff.

I am open to adding the minimum requirement (1 merged PR or document update). It probably can be self attested.

Reply:
If we do put a requirement for 1 PR, it can probably be one over all time, rather than in the last year.

Reply:
Yes

Reply:
@james-jwu How we can deal with contributions that DevStat doesn't count ? E.g. driving meetings, summits, presentations etc. ?

Reply:
@andreyvelich this is already dealt with, see the following line `If a community member has had significant contributions over the past year but is not captured in the dashboard, they will be able to submit an exception form to the Elections Officer(s) who will then review and determine whether this member should be eligible to vote. All exceptions, and the reasons for them, will be recorded in a log that will be available to Steering and the TOC.`

Reply:
An exception can be submitted. This is documented a little further below:
```
If a community member has had significant contributions over the past year but is not captured in the dashboard, they will be able to submit an exception form to the Elections Officer(s) who will then review and determine whether this member should be eligible to vote. All exceptions, and the reasons for them, will be recorded in a log that will be available to Steering and the TOC.
```

Reply:
Added a requirement on minimum 1 merged PR.